android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -0,0 +1,187 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.net.Uri
+import android.util.Log
+import androidx.activity.result.ActivityResultCaller
+import com.solana.mobilewalletadapter.clientlib.protocol.JsonRpc20Client
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationScenario
+import com.solana.mobilewalletadapter.clientlib.scenario.Scenario
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioChooser
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioTypes
+import com.solana.mobilewalletadapter.common.ProtocolContract
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.util.concurrent.CancellationException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+fun LocalAssociationScenario.begin(): MobileWalletAdapterClient {
+    return start().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
```

Comment:
There are some legitimate exceptions that will be thrown by `start()` that should either be handled here (and made friendly), or documented as thrown by this method. For example, if the server drops the connection for any reason before the session is fully established, this will throw a `LocalAssociationScenario.ConnectionFailedException`

Reply:
Did you see my `withScenario` method? I could move the start-catching exceptions to here too, to make it clearer.